nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -1595,6 +1596,48 @@ impl<'a> Planner<'a> {
                 }
                 true
             }
+            ZoneKind::BoundaryNtp => {
+                debug!(
+                    self.log,
+                    "Checking if boundary NTP zone can be shut down"
+                );
+
+                // Find all boundary NTP zones
+                let mut boundary_ntp_zones = std::collections::HashSet::new();
+                for sled_agent in self.inventory.sled_agents.iter() {
```

Comment:
Should this be looking at the current blueprint instead of inventory? If we have an out-of-date inventory, this might show us a boundary NTP zone that has since been expunged (and if it's also still in `inventory.ntp_timesync`, we'd count it as a sync'd zone).